impl/src/main/java/org/jboss/weld/util/Proxies.java

```diff
@@ -56,32 +58,37 @@
 
         private final List<Class<?>> interfaces;
         private final List<Class<?>> classes;
+        private final Map<Class<?>, String> classToPackageMap;
 
         private TypeInfo(Set<? extends Type> types) {
             Comparator<Class<?>> classComparator = Comparator.comparing(Class::getName);
             List<Class<?>> foundInterfaces = new ArrayList<>();
             List<Class<?>> foundClasses = new ArrayList<>();
+            Map<Class<?>, String> classToPackage = new HashMap<>();
 
-            types.stream().forEach(type -> add(type, foundInterfaces, foundClasses));
+            types.stream().forEach(type -> add(type, foundInterfaces, foundClasses, classToPackage));
 
             // sort both collections and create unmodifiable lists
             Collections.sort(foundClasses, classComparator);
             Collections.sort(foundInterfaces, classComparator);
+            // TODO should we re-sort this in a way that non-java[x] class names are first?
```

Comment:
I like the idea...

Reply:
Ok, will add something. Theoretically a smarter comparator should be able to achieve this.
But I think we should only sort interfaces in this case, otherwise we might break something.